testing/trino-testing/src/main/java/io/trino/testing/AbstractTestEngineOnlyQueries.java

```diff
@@ -6605,6 +6605,36 @@ public void testColumnNames()
         assertThat(showCreateTableResult.getColumnNames()).isEqualTo(ImmutableList.of("Create Table"));
     }
 
+    @Test
+    public void testInlineSession()
+    {
+        assertThat(query("WITH SESSION time_zone_id = 'Europe/Wonderland' SELECT current_timezone()"))
+                .failure()
+                .hasMessageContaining("Time zone not supported: Europe/Wonderland");
+
+        assertThat(query("WITH SESSION time_zone_id = 'Europe/Warsaw' SELECT current_timezone()"))
+                .matches("VALUES CAST('Europe/Warsaw' AS varchar)");
+
+        Session session = Session.builder(getSession())
+                .setSystemProperty("time_zone_id", "America/Los_Angeles")
+                .build();
+
+        assertThat(query(session, "WITH SESSION time_zone_id = 'Europe/Warsaw' SELECT current_timezone()"))
+                .matches("VALUES CAST('Europe/Warsaw' AS varchar)");
+    }
+
+    @Test
+    public void testInlineSessionAndSqlFunctions()
+    {
+        assertThat(query("""
+                WITH
+                  SESSION time_zone_id = 'Europe/Warsaw'
+                  FUNCTION foo() RETURNS varchar RETURN current_timezone()
+                SELECT foo()
+                """))
+                .matches("VALUES CAST('Europe/Warsaw' AS varchar)");
+    }
```

Comment:
AbstractTestEngineOnlyQueries gets executed multiple times for different setups of the engine. We don't need to run these tests with all those combinations. Move these to a dedicated class under `io.trino.sql.query`. See other test there as an example.

Reply:
we have `testInlineSqlFunctions` already in that class, was it intentional?

Reply:
I'll follow up with the extraction of testInlineSqlFunctions

Reply:
I'll add commit here